src/node/test/channels.cpp

```diff
@@ -40,6 +47,21 @@ struct NodeOutboundMsg
   NodeMsgType type;
   T authenticated_hdr;
   std::vector<uint8_t> payload;
+
+  std::vector<uint8_t> data() const
+  {
+    std::vector<uint8_t> r;
+    r.insert(r.end(), authenticated_hdr.begin(), authenticated_hdr.end());
+    r.insert(r.end(), payload.begin(), payload.end());
+    return r;
+  }
+
+  std::vector<uint8_t> unauthenticated_data() const
+  {
+    std::vector<uint8_t> r = data();
+    r.erase(r.begin(), r.begin() + 8); // Skip 8 bytes of message type
```

Comment:
As discussed, this is because of the `ChannelMsg` enum included in each node-to-node channel. Can we read it more cleanly?